### PR TITLE
Fix incorrect sorting of ZooLock child nodes

### PR DESCRIPTION
**Pull request by ctubbsii**
* Take into account the new UUID scheme when sorting ZooLock child nodes
* Fix BackupManagerIT missing slash in path to lock node

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/fate/AdminUtil.java [hunk 1]
```diff
@@ -63,7 +63,6 @@ public AdminUtil() {
    *          <code>System.exit(1)</code> on error if true
    */
   public AdminUtil(boolean exitOnError) {
-    super();
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/fate/AdminUtil.java:66]: This was removed by my editor as redundant when I saved the file.

File: core/src/main/java/org/apache/accumulo/fate/AdminUtil.java [hunk 2]
```diff
@@ -293,12 +292,10 @@ private void findLocks(ZooReader zk, final String lockPath,
...
               locks = heldLocks;
+            } else if (lda[0].charAt(0) == 'R' && !sawWriteLock) {
+              locks = heldLocks;
             } else {
-              if (lda[0].charAt(0) == 'R' && !sawWriteLock) {
-                locks = heldLocks;
-              } else {
-                locks = waitingLocks;
-              }
+              locks = waitingLocks;
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/fate/AdminUtil.java:298]: This was converted by my editor as redundant when I saved the file, and is clearly better, so I kept it.

File: test/src/main/java/org/apache/accumulo/test/functional/BackupManagerIT.java [hunk 3]
```diff
@@ -52,13 +52,13 @@ public void test() throws Exception {
...
-        children = writer.getChildren(root + Constants.ZMANAGER_LOCK);
+        String path = root + Constants.ZMANAGER_LOCK;
+        children = ZooLock.validateAndSortChildrenByLockPrefix(path, writer.getChildren(path));
       } while (children.size() != 2);
-      Collections.sort(children);
       // wait for the backup manager to learn to be the backup
       UtilWaitThread.sleep(1000);
       // generate a false zookeeper event
-      String lockPath = root + Constants.ZMANAGER_LOCK + children.get(0);
+      String lockPath = root + Constants.ZMANAGER_LOCK + "/" + children.get(0);
```
Comment (brianloss) [inline test/src/main/java/org/apache/accumulo/test/functional/BackupManagerIT.java:61]: Thanks for catching this. I had run this test and fixed the same bug, which I introduced when I replaced the text with the constant, but in managing multiple pull requests I must have lost the change.

**Review by brianloss — APPROVED** [review]
LGTM

**Pull request merged**